go/arrow/array/record.go

```diff
@@ -180,6 +179,14 @@ func (rec *simpleRecord) SetColumn(i int, arr arrow.Array) (arrow.Record, error)
 	return NewRecord(rec.schema, arrs, rec.rows), nil
 }
 
+func (rec *simpleRecord) TotalBufferSize() uint64 {
```

Comment:
Since you added `TotalRecordSize` in utils, is there actually a need for this separate method? If so, can it just call `TotalRecordSize` and add this to the `arrow.Record` interface?

Reply:
I removed this method as we can't add it to the interface given it will cause the cyclical loop